internal/gengapic/genrest.go

```diff
@@ -0,0 +1,179 @@
+// Copyright (C) 2021  Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gengapic
+
+import (
+	"sort"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/googleapis/gapic-generator-go/internal/errors"
+	"github.com/googleapis/gapic-generator-go/internal/pbinfo"
+	"google.golang.org/genproto/googleapis/api/annotations"
+)
+
+func (g *generator) restClientInit(serv *descriptor.ServiceDescriptorProto, servName string, imp pbinfo.ImportSpec, hasRPCForLRO bool) {
+	p := g.printf
+
+	lowcaseServName := lowerFirst(servName)
+
+	p("// %sRestClient is a client for interacting with %s over REST transport.", lowcaseServName, g.apiName)
+	p("// It satisfies the internal%sClient interface.", servName)
+	p("//")
+	p("// Methods, except Close, may be called concurrently. However, fields must not be modified concurrently with method calls.")
+	p("type %sRestClient struct {", lowcaseServName)
+	p("  host string")
+	p("}")
+	p("")
+	g.restClientUtilities(serv, servName, imp, hasRPCForLRO)
+	g.genRESTMethods(serv, servName)
+}
+
+func (g *generator) genRESTMethods(serv *descriptor.ServiceDescriptorProto, servName string) error {
+	// Clear LROs between services
+	// TODO: handle LRO types shared between rest and grpc clients with a map[bool]
+	g.addMetadataServiceForTransport(serv.GetName(), "rest", servName)
+
+	methods := append(serv.GetMethod(), g.getMixinMethods()...)
+
+	for _, m := range methods {
+		g.methodDoc(m)
+		if err := g.genRESTMethod(servName, serv, m); err != nil {
+			return errors.E(err, "method: %s", m.GetName())
+		}
+		g.addMetadataMethod(serv.GetName(), "rest", m.GetName())
+	}
+
+	sort.Slice(g.aux.lros, func(i, j int) bool {
+		return g.aux.lros[i].GetName() < g.aux.lros[j].GetName()
+	})
+	for _, m := range g.aux.lros {
+		if err := g.lroType(servName, serv, m); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (g *generator) restClientUtilities(serv *descriptor.ServiceDescriptorProto, servName string, imp pbinfo.ImportSpec, hasRPCForLRO bool) {
+	// p := g.printf
```

Comment:
Confirming it's intentional to include this commented out body?

Reply:
This entire function body is commented out to prevent instantiation of rest clients. This is a deliberate choice to keep freedom of movement during initial development, but initially I wasn't thinking along those lines. This commented out cruft is just an abandoned intermediate effort.